hnn_core/network_builder.py

```diff
@@ -503,11 +504,19 @@ def _connect_celltypes(self):
                         self.ncs[connection_name].append(nc)
 
     def _record_extracellular(self):
+        from .extracellular import _calc_potentials_callback
+
         for arr_name, arr in self.net.rec_arrays.items():
             nrn_arr = _ExtracellularArrayBuilder(arr)
-            nrn_arr._build(cvode=_CVODE)
+            nrn_arr._build()
             self._nrn_rec_arrays.update({arr_name: nrn_arr})
 
+            # Attach a callback for calculating the potentials at each time
+            # step. Keep the callbacks in a list so they can be removed later.
+            recording_callback = (_calc_potentials_callback, nrn_arr)
+            _CVODE.extra_scatter_gather(0, recording_callback)
```

Comment:
`extra_scatter_gather` takes tuple or `callback` ?

Reply:
The docs don't state that, just that the argument should be a 'pycallable`. However, the examples they give [here](https://nrn.readthedocs.io/en/latest/python/simctrl/cvode.html?highlight=cvode#CVode.extra_scatter_gather) use the format `(function, argument)`. Who knows if this is the fundamental reason it doesn't work from within a object method.

Reply:
okay understood. For our sanity, can we call that variable `func_and_args` or `recording_callback_and_args` or `callback_and_args` ?

Reply:
+1 for `callback_and_args`

Reply:
Turns out it _can_ take either! If tuple, first element is the function reference, arguments follow. But it turns out not to be needed (e8125d0)